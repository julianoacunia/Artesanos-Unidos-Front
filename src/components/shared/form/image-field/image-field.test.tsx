import * as React from 'react';
import renderer from 'react-test-renderer';
import ImageField from './';

describe('Image Field', () => {
  const dispatch = jest.fn();
  test('should match snapshot and styles', () => {
    const tree = renderer
    .create(
      <ImageField
        input={{
          value: 'test',
          onBlur: () => ({}),
          onChange: () => ({}),
          name: 'test',
          onDragStart: () => ({}),
          onDrop: () => ({}),
          onFocus: () => ({})
        }}
        meta={{
          autofilled: false,
          asyncValidating: false,
          dirty: false,
          dispatch,
          form: 'test',
          initial: 'test',
          invalid: false,
          pristine: false,
          submitting: false,
          submitFailed: false,
          touched: false,
          valid: false,
          visited: false,
        }}
      />
    )
    .toJSON();
      expect(tree).toMatchSnapshot();
  });
});