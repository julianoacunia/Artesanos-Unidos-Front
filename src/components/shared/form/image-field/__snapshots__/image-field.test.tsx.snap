// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Field should match snapshot and styles 1`] = `
<div
  className="container"
>
  <input
    accept="image/png, image/jpeg"
    className="input"
    multiple={false}
    onChange={[Function]}
    type="file"
  />
  <button
    className="containerPreview"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 7v2.99s-1.99.01-2 0V7h-3s.01-1.99 0-2h3V2h2v3h3v2h-3zm-3 4V8h-3V5H5c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2v-8h-3zM5 19l3-4 2 3 3-4 4 5H5z"
      />
    </svg>
  </button>
  <div
    className="imageFieldGroupError"
  />
</div>
`;
